### PR TITLE
Scene. Add Behaviors to the scene class

### PR DESCRIPTION
**Pull request by Dok11**
I had some idea how to use Behaviors with the Scene, but to my regret I did not find such an opportunity.
So, check this PR, please :)

P.S. I made it most similar to how it implemented in the Mesh class.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by Dok11** [comment]
Or, maybe I should to make an empty node and attach the Behavior to them? What is the best practice?

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13674/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13674/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13674/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13674/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13674/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13674/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13674/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13674/merge#BCU1XR#0

**Review by RaananW — COMMENTED** [review]
Behaviors were meant as a stand-alone modules that extend a scene. As a developer you can maintain a reference to the behaviors you have created.
I am trying to understand the use-case behind keeping the references as part of the scene.

We are also trying to keep the scene a little less "centralized"  though it is a hard task :-)

Waiting for your feedback!

**Comment by Dok11** [comment]
I wanted to have a standalone class like GameMode or any other controller that specific for current scene. The Behavior interface allow me to pack my custom behavior and share it between projects or scenes inside a project. And also I can pack the init and detach behavior into the class.
Without Behavior on the scene I can do same things, but it requires me to copy-paste code relative to attach and detach functions and make some relations between a class instance to some scene.

And also look the docs here https://doc.babylonjs.com/features/featuresDeepDive/behaviors

> Babylon.js v3.1 introduced a new component based tools: the behaviors. A behavior is a simple class that can be attached to **a target** where it will provide a specific set of features. Features will be triggered by defined events.

Here talked about a target, not a Node or a Mesh, so I thought here talking about any BabylonJS's entity. So, it is strange for me, why I can't attach my behavior class to the scene :)

**Comment by RaananW** [comment]
But there are no scene behaviors implemented... 

I understand the question - why can't we add behavior to a scene, but - what behavior would you add to a scene? Do you want to use the behaviors to a form of a plugin-system?

File: packages/dev/core/src/scene.ts [hunk 1]
```diff
@@ -5400,4 +5401,78 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
...
+        const index = this._behaviors.indexOf(behavior);
+
+        if (index === -1) {
+            return this;
+        }
+
+        this._behaviors[index].detach();
+        this._behaviors.splice(index, 1);
+
+        return this;
```
Comment (RaananW) [inline packages/dev/core/src/scene.ts:5452]:
```suggestion
        const index = this._behaviors.indexOf(behavior);
        if (index !== -1) {
            this._behaviors.splice(index, 1)[0].detach();
        }
        return this;
```

**Review by RaananW — COMMENTED** [review]
Just a thought of changes to the code (I know it was taken from node).
I am still trying to understand the need, but I'll be happy to merge it, if it makes sense.

**Comment by carolhmj** [comment]
Hello! Thank you for bringing this up. After careful consideration and internal discussion, we have come to the conclusion that it would be best not to implement any changes to Behavior at this time. This is because we want to ensure that any changes we make are fully compatible with the GLTF behaviors extension (https://docs.google.com/document/d/1MT7gL-IEn_PUw-4XGBazMxsyqsxqgAVGYcNeC4Cj_9Q/edit#) when it is finalized. We appreciate your understanding and patience as we work to ensure the best possible outcome for everyone involved.
